tenable/io/v3/vm/tags/schema.py

```diff
@@ -0,0 +1,145 @@
+'''
+Tags API Endpoint Schemas
+'''
+from marshmallow import Schema, fields, validate
+from marshmallow.decorators import post_dump, pre_load, validates_schema
+from marshmallow.exceptions import ValidationError
+
+from tenable.io.v3.base.schema.explore.filters import ParseFilterSchema
+
+
+class TagCategorySchema(Schema):
+    '''
+    Schema for tags category methods
+    '''
+    name = fields.Str(required=True)
+    description = fields.Str()
+
+    @validates_schema
+    def validate_name(self, data, **kwargs):
+        '''
+        Supproting method to validate name string length
+        '''
+        if len(data['name']) >= 127:
+            raise ValidationError('The name must not exceed 127 characters.')
+
+
+class AssetTagSchema(Schema):
+    '''
+    Schema for asset tags methods
+    '''
+    action = fields.Str(data_key='action')
+    assets = fields.List(fields.UUID, data_key='assets')
+    tags = fields.List(fields.UUID, data_key='tags')
+
+
+class CurrentDomainPermissionSchema(Schema):
+    '''
+    Supporting schema for AccessControlSchema class
+    '''
+    id = fields.UUID(required=True)
+    name = fields.Str(required=True)
+    type = fields.Str(
+        validate=validate.OneOf(['USER', 'GROUP']),
+        required=True
+    )
+    permissions = fields.List(
+        fields.Str(
+            validate=validate.OneOf(
+                ['ALL', 'CAN_USE', 'CAN_EDIT', 'CAN_SET_PERMISSIONS']
+            )
+        ),
+        required=True
+    )
+
+    @pre_load
+    def permission_constructor(self, data, **kwargs):
+        '''
+        Supporting method to construct permission constructor
+        '''
+        if isinstance(data, dict) and len(data) == 4:
+            return data
+        elif isinstance(data, tuple) and len(data) == 4:
+            return {
+                'id': data[0],
+                'name': data[1],
+                'type': data[2],
+                'permissions': data[3]
+            }
+        else:
+            return ValidationError('Invalid Permissions definition')
+
+
+class AccessControlSchema(Schema):
+    '''
+    Supporting schema for TagValueSchema class
+    '''
+    current_user_permissions = fields.List(
+        fields.Str(
+            validate=validate.OneOf(
+                ['ALL', 'CAN_USE', 'CAN_EDIT', 'CAN_SET_PERMISSIONS']
+            )
+        ),
+        default=['ALL', 'CAN_EDIT', 'CAN_SET_PERMISSIONS'],
```

Comment:
see if we can use load_default or dump_default

Reply:
Resolved.